src/core/QueryManager.ts

```diff
@@ -1177,38 +1196,40 @@ export class QueryManager<TStore> {
 
         observable = entry.observable;
         if (!observable) {
-          const concast = new Concast([
-            execute(link, operation) as Observable<FetchResult<T>>,
-          ]);
-          observable = entry.observable = concast;
-
-          concast.beforeNext(function cb(method, arg: FetchResult) {
-            if (method === "next" && "hasNext" in arg && arg.hasNext) {
-              concast.beforeNext(cb);
-            } else {
-              inFlightLinkObservables.remove(printedServerQuery, varJson);
-            }
-          });
+          observable = entry.observable = execute(link, operation).pipe(
+            onAnyEvent((event) => {
+              if (
+                event.type !== "next" ||
+                !("hasNext" in event.value) ||
+                !event.value.hasNext
+              ) {
+                inFlightLinkObservables.remove(printedServerQuery, varJson);
+              }
+            }),
+            shareReplay({ bufferSize: 1, refCount: true })
```

Comment:
```suggestion
            shareReplay({ bufferSize: 1, refCount: true })
```

Bump the `bufferSize`

Reply:
Looks like the default is `Infinity` if you omit `bufferSize`, so I went with that default so that all values are replayed.